internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -306,3 +306,37 @@ func (q amsQuotaService) DeleteQuota(subscriptionId string) *errors.ServiceError
 	}
 	return nil
 }
+
+func (q amsQuotaService) IsSubscriptionActive(kafka *dbapi.KafkaRequest) (*bool, error) {
+	billingModel, err := q.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+	if err != nil {
+		return nil, err
+	}
+
+	orgID, err := q.amsClient.GetOrganisationIdFromExternalId(kafka.OrganisationId)
+	if err != nil {
+		return nil, err
+	}
+
+	quotas, err := q.amsClient.GetQuotaCostsForProduct(orgID, billingModel.AMSResource, billingModel.AMSProduct)
+	if err != nil {
+		return nil, err
+	}
+
+	// quota not found, subscription is not valid
+	if len(quotas) == 0 {
+		return &[]bool{false}[0], nil
+	}
+
+	// result will always return one quota for the given org, resource and product for rhosak quotas
+	quota := quotas[0]
+
+	// when a subscription expires in AMS, the allowed value for that quota is set back to 0
+	// if the allowed value is 0 and consumed is greater than this, that denotes that the subscription
+	// has expired and is no longer active.
```

Comment:
What happens to the underlying subscription denoted by the subscriptionID that we save in the database? Does that status of that subscription changes? If so, it is possible to re-use it?

Reply:
This PR is just about updating the `expires_at` column but doesn't do anything to the subscription.
A dedicated reconciler manages the real cluster expiration.
That reconciler will do the following:

- If expires_at< today : delete the cluster (and its subscription)
- If expires_at > today && (expires_at - grace_period) < today : suspend the cluster and leave the subscription untouched

Reply:
I believe that subscription remains untouched, the status will be the same as it was when the subscription is still active. Maybe @ziccardi can confirm. 

> If so, it is possible to re-use it?

If the AMS subscription (SKU allowance) was extended after it expires, then yes we can re-use it. Once the Kafka instance is promoted however, it will create a new subscription and the existing one deleted.

Reply:
The word 'Subscription' can mean the Subscription (SKU allowance) and AMS Subscription resource. It's a bit confusing, if anyone has any other name suggestions to make the separation a bit clearer between these two let me know.

Reply:
Thank you. 

> This PR is just about updating the expires_at column but doesn't do anything to the subscription.

Correct. What I was asking is specifically in relation to the method naming:
```
IsSubscriptionActive(kafkaRequest)
```

Reading the method signature, I thought that it was checking that the Kafka subscription is still active or not? Is that correct?

> I believe that subscription remains untouched, the status will be the same as it was when the subscription is still active. Maybe @ziccardi can confirm.

Right. This is very odd - ideally, I would have expected some fields to change in the subscription to indicate that it has expired. 

This way we precisely know that the subscription for the given Kafka is active or not and not having. By the looks of the current implementation
```
if quota.Allowed() == 0 && quota.Consumed() > quota.Allowed() {
		return &[]bool{false}[0], nil
	}
```
doesn't tell us that a given kafka subscription has expired but it just tells us that for a given billing model and product, there are some expired subscriptions, correct?

Reply:
Correct. What the method check id if the user is still entitled to have the instance, hence, he still has an active subscription. As Jameel said before, the `subscription` name is reused in many different topics so it can be confusing.

Maybe we should rename the function to '`IsEntitled` or something similar? WDYT?

Reply:
> As Jameel said before, the subscription name is reused in many different topics so it can be confusing.

I only saw Jameel's reply after I posted my comment above, but I agree with what's said there. The name is confusing.

> Maybe we should rename the function to 'IsEntitled or something similar? WDYT?

+1 on renaming it. What do you think of `IsSKUEntitlementStillActive(...)` ?

Reply:
+1 for me

Reply:
Thanks for the suggestions! Not sure about `SKU` because that term isn't really used in the quota-management-list 🤔  Would `IsQuotaEntitlementActive` make sense? the method call would then be `quotaService.IsQuotaEntitlementActive()`? 

We also need to clarify this separation in our docs, I can update them in this PR as well.

Reply:
+1 from me @JameelB Thanks.

Agreeing on the docs modification as well.